kernel/aster-nix/src/vdso.rs

```diff
@@ -197,51 +231,90 @@ impl Vdso {
         }
     }
 
-    /// Return the vdso vmo.
+    /// Return the VDSO vmo.
     fn vmo(&self) -> Arc<Vmo> {
         self.vmo.clone()
     }
 
-    fn update_instant(&self, instant: Instant, instant_cycles: u64) {
-        self.data.lock().update_instant(instant, instant_cycles);
+    fn update_high_res_instant(&self, instant: Instant, instant_cycles: u64) {
+        self.data
+            .lock()
+            .update_high_res_instant(instant, instant_cycles);
 
         // Update begins.
+        let seq_lock = SEQ_LOCK.lock();
         self.vmo.write_val(0x80, &1).unwrap();
         self.vmo.write_val(0x88, &instant_cycles).unwrap();
-        for clock_id in ALL_SUPPORTED_CLOCK_IDS {
+        for clock_id in HIGH_RES_CLOCK_IDS {
             self.update_vmo_instant(clock_id);
         }
         // Update finishes.
         self.vmo.write_val(0x80, &0).unwrap();
```

Comment:
Do not attempt to acquire `Mutex` (used inside `Vmo`) while holding a `SpinLock` (`SEQ_LOCK` here).